docs/developer-guide/syntax-reference.md

```diff
@@ -254,13 +254,17 @@ SELECT ARRAY[s1.colA, s1.colB, s1.colC] FROM s1 EMIT CHANGES;
 MAP<KeyType, ValueType>
 ```
 
+!!! note
+    The `MAP` type is only supported for value columns, not key columns, as map keys
+    may lead to unexpected behavior due to inconsistent serialization.
+    This restriction includes nested types containing maps as well.
+
 !!! note
 		The `DELIMITED` format doesn't support maps.
 
 ksqlDB supports fields that are maps. A map has a key and value type. All
 of the keys must be of the same type, and all of the values must be also
-be of the same type. Currently only `STRING` keys are supported. The
-value type can be any valid SQL type.
+be of the same type.
```

Comment:
are you sure this is true? I thought there were some assumptions baked in at the ksql layer (not the serde layer) about string keys.

Reply:
That's funny, the reason I made this change was because I remembered https://github.com/confluentinc/ksql/pull/6182 which removed all the assumptions that were previously baked in at non-serde layers. However. I forgot that the PR didn't actually add support for protobuf maps with non-string keys, even though we in theory could. So ultimately you're right ( for the wrong reason ;) ). I'll put the statement back in.

Reply:
Thanks for the catch!